### PR TITLE
Change close button appearence to be closer to native

### LINKED ISSUE #270: Use more subtle tab close button
Here is Firefox native tab close button
image

Compare it to Tab Center Redux much heavier one:
image

### LINKED ISSUE #350: Implement new defaultFavicon.svg with context-fill
The existing `defaultFavicon.svg` is obsolete and should be replaced with the new photon version.

See https://bugzilla.mozilla.org/show_bug.cgi?id=1395732

example

### PR DESCRIPTION
**Pull request by ariasuni**
fix #270 
fix #350 

Close button is smaller, closer to the right of the sidebar and less rounded.

Before:
screenshot_20180331_003536

After:
screenshot_20180331_003551

### CODE REVIEW

**Comment by eoger** [comment]
This looks like a good endeavor to pursue, I think we might as well go all the way and use Firefox's .svg and measurements: https://searchfox.org/mozilla-central/rev/42b34ba1961e37e0d2236deafdd126a0ba21b9ec/toolkit/themes/shared/close-icon.inc.css#8

**Comment by ariasuni** [comment]
I updated Photon icons from [here](https://design.firefox.com/icons/viewer/). Note that now, the close icon is black as opposed to some shade of gray in Firefox, but since I do not know well either the CSS of Tab Center Redux nor the CSS properties involved, I’ll probably let that to another person.

**Comment by ariasuni** [comment]
I re-did my work to match current version of Tab Center Redux. It seems to me that the current appearance of the close button matches the rest of the extension, while being reasonably similar to the Firefox tab bar.

**Comment by eoger** [comment]
Apologies for the delay, I've tried the PR and it's looking good.
However, I think the color of the X should match the one present in the native Firefox tabs.

**Comment by ariasuni** [comment]
I tried to use the same code that in the link of [your previous comment](https://github.com/eoger/tabcenter-redux/pull/328#issuecomment-377639682), then made a few searches and tried different ways to change the color used to fill the SVG, with no success.

Basically, [we can’t do it this way](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill#Using_context-fill), because [we can’t use `-moz-content-properties`](https://developer.mozilla.org/en-US/docs/Web/CSS/-moz-context-properties). So I edited the SVG. :woman_shrugging: I changed the `fill` value of the SVG to the color value used by the light Firefox theme.

Note that in the future, improving the icons for dark theme will simply be a matter of copying each icon and changing the `fill` attribute of the SVG, which is very easy and doesn’t bloat the extension since the SVGs are so tiny.

**Comment by ariasuni** [comment]
Are you satisfied with the proposed changes?

**Comment by eoger** [comment]
Sorry I'm currently in a place with no computer, I'll try to get back
to you ASAP.We use a hack to get "fill" working on svgs, search for "filter" in the
tabcenter.css

On Sat, May 12, 2018, at 2:29 PM, Mélanie Chauvel (ariasuni) wrote:
> Are you satisfied with the proposed changes?


> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[1], or mute the thread[2].> 

Links:

  1. https://github.com/eoger/tabcenter-redux/pull/328#issuecomment-388573999

**Comment by Keith94** [comment]
One flaw I should mention is how the close button retains the `:active` background color even though it is not being hovered. For example, click and hold on the close button and move your cursor to the left on the tab. At this point, the background should turn transparent while not being hovered, but it stays the same.

**Comment by ariasuni** [comment]
@Keith94 Right, I noticed that Tab Center Redux’s behavior was slightly different than Firefox native one. So in the last commit, I changed the CSS to match Firefox’s behavior in this regards.

**Comment by ariasuni** [comment]
Oops, it’s a mistake. I’ll fix that.

**Comment by Keith94** [comment]
@ariasuni Since you are updating some photon icons, do you mind looking at #350?

**Comment by Keith94** [comment]
As well, the native tab close button has a transparent background if it's not hovered. Perhaps Tab Center should reflect that, too.

**Comment by ariasuni** [comment]
I got the icon from [Firefox’s source code](https://searchfox.org/mozilla-central/source/toolkit/themes/shared/places/defaultFavicon.svg), because it’s not on [Photon icons website](https://design.firefox.com/icons/viewer/). I modified `tab.js` to add or remove a class — `.default-favicon` — so that I can change default favicon’s color without affecting other favicons.

I removed `background-size` when only tab is hovered, so that tab close icon has a background only when the icon is hovered, to match Firefox’s behavior. @eoger, tell me if you’re okay with this change, otherwise I’ll revert it.

**Comment by Keith94** [comment]
Nice, I appreciate all your efforts! I just noticed a couple issues.

In your screenshot, about:debugging should be using the default favicon. As well, the default favicon itself looks grayish, but elsewhere it looks black (or white depending on theme).

firefox_2018-05-19_09-54-23

**Comment by ariasuni** [comment]
I simplified and fixed my code so that at first load, `about:debugging` favicon is displayed like before.

I removed the CSS that made the default favicon theme dependent, since it’s always loaded on a white background and to match how other `about:` pages favicons are displayed.

**Comment by Keith94** [comment]
Thanks!

>I removed the CSS that made the default favicon theme dependent, since it’s always loaded on a white background and to match how other about: pages favicons are displayed.

There is one issue in that the favicon is hard to see against the _dark theme_ background; however, any dark favicons all share this same problem... not sure if there's a good solution.

**Comment by ariasuni** [comment]
I fixed it. It adds a `fill-color` class when favicon is default favicon or is internal to the navigator (`chrome://` urls) and finish by `.svg` (to avoid recoloring `.png`), otherwise remove it.

File: src/sidebar/tabcenter.css [hunk 1]
```diff
@@ -813,8 +834,7 @@ body.dark-theme {
     --tab-border-color: #38383D;
     --searchbox-background: rgb(71, 71, 73);
     --primary-text-color: rgb(255, 255, 255);
-
-    --close-button-color:  rgba(238, 238, 241, .05);
+    --icon-color: #bebebe;
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Can we rename this to `--icons-fill-color`?

File: src/sidebar/tabcenter.css [hunk 2]
```diff
@@ -826,31 +846,3 @@ body.dark-theme {
 body.dark-theme #newtab-menu {
     color: black;
 }
-
-body.dark-theme #searchbox-icon {
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: The dark theme looked fine to me after testing your patch, but please double check this didn't break anything.

File: src/sidebar/tabcenter.css [hunk 3]
```diff
@@ -702,7 +723,7 @@ body[platform="win"] #searchbox-input:placeholder-shown {
 .tab-close:hover {
     background-color: var(--close-button-hover) !important;
 }
-.tab-close:active {
+.tab-close:hover:active {
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: 👌

File: src/sidebar/tabcenter.css [hunk 4]
```diff
@@ -680,17 +699,19 @@ body[platform="win"] #searchbox-input:placeholder-shown {
 .tab-close {
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Almost pixel perfect, but the size of the **X** was bothering me:

comp
The 40x40px gray hover background is the right size however!

File: src/sidebar/img/find.svg [hunk 5]
```diff
@@ -1,22 +1,6 @@
-<?xml version="1.0"?>
```
Comment (eoger) [inline src/sidebar/img/find.svg]: This new icon doesn't align with the other sidebar ones (bookmarks, history, synced tabs). Could we restore the old one? However, I like the removal of #standard/#inverted :)

File: src/sidebar/img/glyph-pin-pinned-12.svg [hunk 6]
```diff
@@ -1,24 +1,6 @@
-<?xml version="1.0"?>
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (eoger) [inline src/sidebar/img/glyph-pin-pinned-12.svg]: Are all of these new icons licensed with MPL2/in the Firefox source code?

File: src/sidebar/tabcenter.css [hunk 7]
```diff
@@ -577,13 +582,24 @@ body[platform="win"] #searchbox-input:placeholder-shown {
...
 }
 
+.tab-icon.fill-color {
+    -moz-context-properties: fill;
+    fill: var(--icon-color);
+    filter: url("img/filters.svg#fill");
+}
+
+#tablist:not(.shrinked) .tab-icon.fill-color {
+    fill: #5a5b5c;
```
Comment (eoger) [inline src/sidebar/tabcenter.css]: Do we need a different color in these circumstances?

File: src/sidebar/tab.js [hunk 8]
```diff
@@ -165,6 +163,11 @@ SideTab.prototype = {
     toggleClass(this.view, "hidden", !show);
   },
   _setIcon(favIconUrl) {
+    if (favIconUrl.startsWith("chrome://") && favIconUrl.endsWith(".svg")) {
+      this._iconView.classList.add("fill-color");
```
Comment (eoger) [inline src/sidebar/tab.js]: Let's rename the class to something more explicit, like `chrome-icon`.

File: src/sidebar/tabcenter.css (hunk 7 above)
Comment (eoger) [reply src/sidebar/tabcenter.css]: I've also noticed that this triggers a bug where the colors are different:

screen shot 2018-05-21 at 12 47 04 pm

File: src/sidebar/img/glyph-pin-pinned-12.svg (hunk 6 above)
Comment (ariasuni) [reply src/sidebar/img/glyph-pin-pinned-12.svg]: They are from here: https://design.firefox.com/icons/viewer/

**Comment by ariasuni** [comment]
It should be all good now!

**Comment by Keith94** [comment]
This is how my search bar looks compared to the bookmarks sidebar one on Windows 10.

firefox_2018-05-23_16-24-06
firefox_2018-05-23_16-23-03

It could use some minor polish (e.g. no italics) but otherwise looks good.

**Comment by ariasuni** [comment]
The icon could be tweaked a bit, but it seems the intent was to make it look like the URL bar, not like the native search field used in other panes. We could probably remove italic for Windows for consistency since it’s not used in others bars, though.

Also, we don’t want this pull request to become too big, at least I don’t. So, let’s focus on tabs (favicons, close buttons) for this one.

**Comment by eoger** [comment]
Merged in https://github.com/eoger/tabcenter-redux/commit/b99e90507b787be11320f2943bde5ed24ab0f0b1.
Thank you for your patience and dedication!
What we are experiencing here with the search box is probably platform differences (looks fine to me on macOS), we should address this in another PR.

**Pull request closed without merging**